config/v1/types_descheduling.go

```diff
@@ -0,0 +1,45 @@
+package v1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Descheduler holds cluster-wide config information to run the Kubernetes Descheduler
+// and influence its placement decisions. The canonical name for this config is `cluster`.
+type Descheduler struct {
```

Comment:
Seems like if this is going to become a top level configuration resource, it should be a stanza in our existing scheduler configuration.

Reply:
The scheduler(+operator) doesn't handle anything to do with the descheduler, and vice-versa, so coupling them doesn't seem appropriate.

Reply:
> The scheduler(+operator) doesn't handle anything to do with the descheduler, and vice-versa, so coupling them doesn't seem appropriate.

The choice of where configuration lives is not made for developer convenience.  config.openshift.io is about how a cluster-admin configures his cluster.  Regardless of how it works under the covers, all scheduling configuration is related.  If  this APIs  lives in config.openshift.io (and I doubt it), it is part of scheduling configuration no matter what your operator does with it.